lib/new_relic/agent/http_clients/httpclient_wrappers.rb

```diff
@@ -57,12 +57,16 @@ def uri
         end
 
         def [](key)
-          request.headers[key]
+          headers[key]
         end
 
         def []=(key, value)
           request.http_header[key] = value
```

Comment:
I'm tempted to move `[]` and `[]=` into the base class and just require every subclass to define a `headers` method. This seems to be the only method that wouldn't work with that, explicitly doing something other than `headers[key] = value`.

Reply:
I think there's a lot of similarities between the `[]` and `[]=` methods between the subclasses, but there's some uniqueness in typheous, httpclient, and net_http versions of `[]=` that make me hesitant to switch them over. 

There should now be a `headers` method in each subclass that accesses a hash of headers. In some libraries the hash of headers is not the same object where things are supposed to be added (ex. http.rb, net_http). 

Thoughts?

Reply:
Fully 3 Ruby HTTP clients being incompatible with the approach is definitely a strong reason to not go with it. Sounds good.

Having the common `headers` method for getting values is great at least. Nice!